docs: :building_construction: interface design for `convert` (#63)
seedcase-project/zen-do: docs/design/interface/cli.qmd

    # deposit: Deposit = find_deposit(deposits=deposits, metadata=metadata)
    pretty_print(deposit)
```

## {{< var planned >}} `convert`

``` {.bash filename="Terminal"}
zen-do convert [METADATA_FILE] --to [FORMAT]
```

There are many different file formats and standards that contain similar
or even identical metadata, but that are structured slightly differently

> or are in different file formats. The `convert` command helps to keep

Reviewer: ```suggestion
or even identical metadata, but that are structured slightly differently.
The `convert` command helps to keep
```
Repeated from the beginning of the sentence

  [CodeMeta](https://codemeta.github.io/) standard. It contains similar
  metadata fields to the `.zenodo.toml` file.
- {{< var planned >}} `_quarto.yml`: This file is used to build
  [Quarto](https://quarto.org/) documents. Like `pyproject.toml`, it
  also contains similar metadata fields to the `.zenodo.toml` file, such
  as the title, description, and authors.
- {{< var planned >}} `DESCRIPTION`: This is a typical file found in R
  projects. It is necessary for developing and building R packages. As
  with `pyproject.toml`, it contains similar metadata fields to the
  `.zenodo.toml` file, such as the package name, description, authors,
  and license.

The `--to` argument can be a single format or an array of formats,
allowing users to convert to multiple formats in one command.

```python
from cyclopts import App, Parameter
from pathlib import Path
from typing import Annotated
from enum import StrEnum

class Format(StrEnum):
    zenodo_json = ".zenodo.json"
    citation_cff = "CITATION.cff"
    pyproject_toml = "pyproject.toml"
    quarto_yml = "_quarto.yml"
    description = "DESCRIPTION"

@app.command()
def convert(

>         metadata_file: Path = Path(".zenodo.toml"),

Reviewer: Do we want to separate this from the others by `/, *` as in the `get` command?

>         metadata_file: Path = Path(".zenodo.toml"),

Author: ```suggestion
        metadata_file: Path = Path(".zenodo.toml"),
        /,
        *,
```

        to: Annotated[list[Format], Parameter(consume_multiple=True)],
        verbose: bool = False
    ) -> None:
    """Convert a metadata file, e.g. `.zenodo.toml`, to other formats.

    Args:
        metadata_file: The path to the metadata file to convert.
        to: The formats to convert the metadata file to. Can be a single format
            or an array of formats.
        verbose: Whether to print a log of the actions done.
    """
    # Read metadata file (checking if the URN exists or not).
    # metadata: Metadata = read_metadata(path=metadata_file)
    # Convert to the given formats

>     # formats: list[str] = convert_to_formats(metadata, to)

Author: ```suggestion
    # formats: list[str] = convert_to_formats(metadata, to)
    # TODO: Consider how updating existing files will work, need to read other formats first?
```

>     # output_path: list[Path] = write_formats(formats)

Reviewer: Do you prefer this (iterating over the formats twice) to iterating over the formats once, and in each iteration doing the conversion and writing together for that format?

I'm just thinking that we will have to map from the format's name to a conversion function and an output path, but we only need one map if we bundle the conversion and writing together.

Or do we care about what happens when one conversion fails? Should that prevent all output files from being written or let the good ones through? (Not terribly important, but it informs the design, so I'm wondering.)

Author: For I/O tasks, those should be stand-alone functions, makes it easier to test and makes it more flexible to pre-writing changes.

And if writing fails, don't cause a failure, just output a warning/log.

And I think the writing might have to be two-step since we don't want to overwrite existing metadata in a file and only change the fields that are the same. I'll add a TODO item for now to think about it.
